beets/dbcore/db.py

```diff
@@ -308,7 +307,7 @@ class Model(ABC):
     are subclasses of `Sort`.
     """
 
-    _queries: Dict[str, Type[Query]] = {}
+    _queries: Dict[str, Type[FieldQuery]] = {}
```

Comment:
I think this should remain unchanged, at least if we keep the `Query` base class around.

Reply:
My thinking here was that `Query` was required in order to accommodate this one single `PlaylistQuery` that was not a `FieldQuery`. Seeing that all values in `Album._queries` and `Item._queries` now subclass a more specific `FieldQuery`, I aligned it accordingly.

Do you reckon it may be worth enforcing `FieldQuery` here, so that maintainers and plugin developers building on this have slightly more specific requirements/guidelines over how their queries should be implemented?

Reply:
My thinking was that `db` only really uses the interface defined by `Query`, nothing added by `FieldQuery` on top.

So either, `Query` and `FieldQuery` remain distinct, in which case my comment from above applies.

If we decide that there's really nothing more basic than a `FieldQuery`, we should just remove one of them and simplify the class hierarchy. On first thought, I wouldn't do this, since conceptually there might well be different types of queries, and as far as I can tell, this doesn't really increase the complexity of our code very much.

Does this make sense? I might very well be missing something, it's been a while since I've looked at the `dbcore` code in any depth.

Reply:
> My thinking was that db only really uses the interface defined by Query, nothing added by FieldQuery on top.

I haven't adjusted what the db expects here, for example see `Database._fetch` definition which has `Query`

```python
    def _fetch(
        self,
        model_cls: Type[AnyModel],
        query: Optional[Query] = None,
        sort: Optional[Sort] = None,
    ) -> Results[AnyModel]:
```

The change above adjusts `Model._queries`, which defines the type for `Item._queries` and `Album._queries`:

```python
class Item(LibModel):
    ...

    _queries = {"singleton": SingletonQuery}
```

This is also the dictionary that keeps named queries like `PlaylistQuery` once the plugins are parsed.

I have just attempted to change this to `_queries: Dict[str, Query]` and I found this made `mypy` upset in `queryparse.py`:
![image](https://github.com/beetbox/beets/assets/16212750/ee0ec925-a514-41ea-b7f0-b161fee5ea56)

This is most likely why I adjusted it in the first place

Reply:
I see the problem now: `db` does indeed not require this (anyway, it does not even access the `_queries` attribute), but `queryparse` uses the `db` default queries to initialize the `query_classes`.
So, I guess your change is fine. It still feels a little rough at the edges, because `Model._queries` used to be the non-field-queries, which are now all `FieldQuery` in disguise. I guess we can just keep these things as is, and figure out a better way if either non-field-queries need to return at some point, or if we decide to completely remove this distinction.